distributed/diagnostics/plugin.py

```diff
@@ -361,20 +364,111 @@ class PackageInstall(WorkerPlugin, abc.ABC):
     PipInstall
     """
 
-    INSTALLER: ClassVar[str]
+    LOCK: ClassVar[asyncio.Lock | None] = None
+    WORKER_PLUGIN_CLASS: ClassVar[type]
 
+    installer: _PackageInstaller
     name: str
-    packages: list[str]
-    restart: bool
+    restart_workers: bool
+    _scheduler: Scheduler
 
     def __init__(
         self,
-        packages: list[str],
-        restart: bool,
+        installer: _PackageInstaller,
+        restart_workers: bool,
     ):
-        self.packages = packages
+        self.installer = installer
+        self.restart_workers = restart_workers
+        self.name = f"{self.installer.INSTALLER}-install-{uuid.uuid4()}"
+
+    async def start(self, scheduler: Scheduler) -> None:
+        self._scheduler = scheduler
+
+        if PackageInstall.LOCK is None:
+            PackageInstall.LOCK = asyncio.Lock()
+
+        async with PackageInstall.LOCK:
+            if not self._is_installed(scheduler):
+                logger.info(
+                    "%s installing the following packages on the scheduler: %s",
+                    self.installer.INSTALLER,
+                    self.installer.packages,
+                )
+                self._set_installed(scheduler)
+                self.installer.install()
```

Comment:
```suggestion
                self.installer.install()
                self._set_installed(scheduler)
```
This may matter if install() occasionally raises?